internal/restorer/filerestorer.go

```diff
@@ -301,10 +302,15 @@ func (r *fileRestorer) downloadPack(ctx context.Context, pack *packInfo) {
 
 }
 
-func (r *fileRestorer) loadBlob(rd io.Reader, blobID restic.ID, length int) ([]byte, error) {
+func (r *fileRestorer) loadBlob(rd io.Reader, blobID restic.ID, length int, buf []byte) ([]byte, error) {
 	// TODO reconcile with Repository#loadBlob implementation
 
-	buf := make([]byte, length)
+	switch {
+	case cap(buf) < length:
+		buf = make([]byte, length)
+	case len(buf) != length:
+		buf = buf[:length]
```

Comment:
I'd call `buf = buf[:length]` unconditionally. reslicing is probably cheaper than / just as expensive as checking the buffer length.

Reply:
is changed.